aten/src/ATen/native/native_functions.yaml

```diff
@@ -4517,6 +4539,7 @@
   dispatch:
     CPU: legacy::cpu::_th_nonzero
     CUDA: legacy::cuda::_th_nonzero
+    HAMMERBLADE: legacy::cuda::_th_nonzero
```

Comment:
The yaml file is case sensitive, so `HAMMERBLADE` will case a building failure. Also, why we are calling `legacy::cuda::_th_nonzero` for HammerBlade?